File: app/models/kanji.rb
class Kanji < ApplicationRecord
  include LevelAble, JsonAble

  DEFAULT_ATTRS = [:id, :kanji_text, :radical, :kanji_pron, :on_pron, :kun_pron, :mean, :memo, :examples]

  has_many :lessons, class_name: 'LessonDetail', as: :topicable, dependent: :destroy
  has_many :example_ables, as: :exampleable, dependent: :destroy
  has_many :examples, through: :example_ables, source: :example

  serialize :radical
  serialize :kanji_pron
  serialize :kun_pron
  serialize :on_pron
  serialize :memo
end

File: front/src/app/classes/kanji.ts
import { IKanji } from '../interfaces/ikanji';
import { IExample } from '../interfaces/iexample';

export class Kanji implements IKanji {
  
  public id: number;
  public kanji_text: string;
  public kanji_pron: string[];
  public kun_pron: string[];
  public on_pron: string[];
  public mean: string;
  public memo: string[];
  public radical: { label: string; value: string }[];
  public examples: IExample[];

  constructor(
    object: {
      id: number;
      kanji_text: string;
      kanji_pron: string[];
      kun_pron: string[];
      on_pron: string[];
      mean: string;
      memo: string[];
      radical: string[];
    }
    ) {
    Object.assign(this, object)
  }
}